pkgs/epnix/python-modules/softioc/default.nix

```
@@ -0,0 +1,40 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchgit,
+  setuptools,
+  setuptools_scm,
+  epicscorelibs,
+  pvxslibs,
+  epicsdbbuilder,
+  epnixLib,
+}:
+buildPythonPackage rec {
+  pname = "softioc";
+  version = "4.5.0";
+
+  pyproject = true;
+
+  src = fetchgit {
+    url = "https://github.com/DiamondLightSource/pythonSoftIOC.git";
+    rev = version;
+    fetchSubmodules = true;
+    hash = "sha256-JXfFkA3MzipqUw0riMTZmgCP9qe4Tfj8vZaFBwqoO+c=";
+  };
+
+  # Set correct verion instead of automatically detected version:
+  postPatch = ''
+    awk -i inplace "/__version__/ && !x {print; print \"__version__ = '${version}'\"; x=1; next} 1" setup.py
+  '';
+
+  nativeBuildInputs = [setuptools setuptools_scm];
+
+  propagatedBuildInputs = [epicscorelibs epicsdbbuilder pvxslibs];
```

```suggestion
  build-system = [setuptools];
  dependencies = [setuptools epicscorelibs epicsdbbuilder pvxslibs];
```

It seems setuptools is also used at runtime, by running `pythonSoftIOC`